auparse/interpret.c

```diff
@@ -2876,6 +2886,20 @@ int auparse_interp_adjust_type(int rtype, const char *name, const char *val)
 		(rtype == AUDIT_ADD_GROUP || rtype == AUDIT_GRP_MGMT ||
 			rtype == AUDIT_DEL_GROUP))
 		type = AUPARSE_TYPE_GID;
+	else if (rtype == AUDIT_TRUSTED_APP) {
+		if (val[0] == '"')
+			type = AUPARSE_TYPE_ESCAPED;
+		else if (strcmp(name, "id") == 0)
```

Comment:
Why are we singling out "id" as needing interpretation? If we do that one, then we need to do more. Perhaps just detecting escaped for trusted app records is all that should be done. It should somehow call lookup_type otherwise.

Reply:
This is prevents possible unescaping for integer values for id-like fields.

Reply:
Right. But what I'm thinking is that it should be something like 
```
else if (rtype == AUDIT_TRUSTED_APP) {
        if (val[0] == '"')
                type = AUPARSE_TYPE_ESCAPED;
        else
                type = lookup_type(name);
}
```
This way it looks up more than just "id".

Reply:
No, thanks.